jupyterbook/content/code_gallery/data_management_notebooks/2023-03-20-Reading_and_writing_zarr.ipynb

```diff
@@ -0,0 +1,1681 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "29370978",
+   "metadata": {},
+   "source": [
+    "# Reading and writting zarr files with xarray\n",
+    "\n",
+    "\n",
+    "Created: 2023-03-20\n",
+    "\n",
+    "\n",
+    "\n",
+    "[The zarr format](https://zarr.readthedocs.io/en/stable/) is a file storage based specification for chunked, compressed, N-dimensional arrays. The format is based on an open-source specification and its main goal is to make cloud data read/write a bit easier and more effective.\n",
+    "\n",
+    "The main propblems in data storage are:\n",
+    "\n",
+    "1. Read/write data that is larger than memory\n",
+    "1. Being able to parallelize computations\n",
+    "1. Reduce the I/O botteneck\n",
+    "1. Compression\n",
+    "1. Speed\n",
+    "\n",
+    "One solution is to use a chunked\\* parallel computing framework and a chunked parallel storage library. Zarr helps us with the latter.\n",
+    "\n",
+    "In this example we will load an ocean model data, stored as netCDF and served via THREDDS, subset it and save as zarr. Let's start by saving a single time step for the surface layer temperature and salinity.\n",
+    "\n",
+    "\n",
+    "\\* Many data formats can take advantage of storing the data in chunks for faster access, the zarr approach is differnt in that each chunk is a different object in cloud storage, making them better for parallel access. The chunks can be compressed to reduce their size and improve cloud performance even further. Zarr has a nice tutorial on how to balance chunk size for performance. Check it out: https://zarr.readthedocs.io/en/stable/tutorial.html#chunk-optimizations."
```

Comment:
`differnt` -> `different`

Reply:
What's up with my spell checker? I think the jupyter update  broke it :-/

Fixed in the last commit.